### ISSUE TITLE
Support for Dynamic Port Allocation

### ISSUE THREAD

**Issue by alexander-verharen-qc** [terraform-google-modules/terraform-google-cloud-nat#89]
### TL;DR

Google and Terraform support dynamic port allocation, while this module does not yet support it. 

The following new variables are needed:

[max_ports_per_vm](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_router_nat#max_ports_per_vm) - (Optional) Maximum number of ports allocated to a VM from this NAT. This field can only be set when enableDynamicPortAllocation is enabled.

[enable_dynamic_port_allocation](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_router_nat#enable_dynamic_port_allocation) - (Optional) Enable Dynamic Port Allocation. If minPortsPerVm is set, minPortsPerVm must be set to a power of two greater than or equal to 32. If minPortsPerVm is not set, a minimum of 32 ports will be allocated to a VM from this NAT config. If maxPortsPerVm is set, maxPortsPerVm must be set to a power of two greater than minPortsPerVm. If maxPortsPerVm is not set, a maximum of 65536 ports will be allocated to a VM from this NAT config. Mutually exclusive with enableEndpointIndependentMapping.

### Terraform Resources

```markdown
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_router_nat#enable_dynamic_port_allocation
```

### Detailed design

```markdown
enableDynamicPortAllocation can only be set to true if enableEndpointIndependentMapping is false.
```

### Additional information

Google provider version 4.5 or greater

**Comment by alexander-verharen-qc** [comment]
sorry - duplicate issue

**Comment by alexander-verharen-qc** [comment]
Closing
